### repo: C:\Users\diaso\tinker-cookbook @ 5e6b47c6bce29084bfc4ec500e98fcc93da3a4f0
modules: tinker_cookbook
-- repo-wide (vs HEAD, staged+unstaged) --
diff --git a/.gitignore b/.gitignore
index e75911d..e9c3f40 100644
--- a/.gitignore
+++ b/.gitignore
@@ -19,3 +19,4 @@ tinker_cookbook/docs/*
 tinker_cookbook/sandbox/*
 tinker_cookbook/assets/*
 tinker_cookbook/recipes/code_rl/code_generation
+tinker_cookbook/recipes/code_rl/test_wandb_visual.py
diff --git a/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/code.diff b/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/code.diff
deleted file mode 100644
index 9e35d9b..0000000
--- a/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/code.diff
+++ /dev/null
@@ -1,176 +0,0 @@
-### repo: C:\Users\diaso\tinker-cookbook @ 1e2cddc9fc832ce66310c327e94603588f74216c
-modules: tinker_cookbook
--- repo-wide (vs HEAD, staged+unstaged) --
-diff --git a/tinker_cookbook/recipes/code_rl/gemini_cli_code_env.py b/tinker_cookbook/recipes/code_rl/gemini_cli_code_env.py
-index e4973c3..d9c6b50 100644
---- a/tinker_cookbook/recipes/code_rl/gemini_cli_code_env.py
-+++ b/tinker_cookbook/recipes/code_rl/gemini_cli_code_env.py
-@@ -118,6 +118,7 @@ class CodeEnv_Gemini(ProblemEnv):
-         format_coef: float = 0.1,
-         reward_timeout: int = 6,
-         code_qual_every_n: int = 10,
-+        gemini_weight: float = 0.2
-     ):
-         super().__init__(renderer, convo_prefix, format_coef=format_coef)
-         self.problem = problem
-@@ -125,7 +126,7 @@ class CodeEnv_Gemini(ProblemEnv):
-         self.reward_timeout = reward_timeout
-         self.code_qual_every_n = code_qual_every_n
-         self.time_step = 0
--        # self.code_qual_score = 0
-+        self.gemini_weight = gemini_weight
- 
-     def get_question(self) -> str:
-         return self.problem
-@@ -186,7 +187,7 @@ class CodeEnv_Gemini(ProblemEnv):
-                 code_qual_score = 0.0
- 
-         # Total reward includes code quality score
--        total_reward = self.format_coef * (format_score - 1.0) + correct_score + code_qual_score
-+        total_reward = self.format_coef * (format_score - 1.0) + correct_score + code_qual_score * self.gemini_weight
- 
-         # Log summary at top of page (CSS ordering puts .lt-summary at top)
-         logtree.log_summary([
-@@ -244,6 +245,7 @@ class DeepcoderDataset_Gem(RLDataset):
-         format_coef: float = 0.1,
-         reward_timeout: int = 6,
-         code_qual_every_n: int = 10,
-+        gemini_weight: float = 0.2
-     ):
-         self.ds = _load_deepcoder_split(split)
-         if split == "train":
-@@ -255,6 +257,7 @@ class DeepcoderDataset_Gem(RLDataset):
-         self.format_coef = format_coef
-         self.reward_timeout = reward_timeout
-         self.code_qual_every_n = code_qual_every_n
-+        self.gemini_weight = gemini_weight
- 
-     def __len__(self) -> int:
-         return (len(self.ds) + self.batch_size - 1) // self.batch_size
-@@ -293,6 +296,7 @@ class DeepcoderDataset_Gem(RLDataset):
-                 format_coef=self.format_coef,
-                 reward_timeout=self.reward_timeout,
-                 code_qual_every_n=self.code_qual_every_n,
-+                gemini_weight=self.gemini_weight
-             ),
-             num_envs=group_size,
-             dataset_name="deepcoder",
-@@ -310,6 +314,7 @@ class DeepcoderDatasetBuilder_Gem(RLDatasetBuilder):
-     format_coef: float = 0.1
-     reward_timeout: int = 6
-     code_qual_every_n: int = 10
-+    gemini_weight:float =0.2
- 
-     async def __call__(self) -> tuple[DeepcoderDataset_Gem, DeepcoderDataset_Gem]:
-         tokenizer = get_tokenizer(self.model_name_for_tokenizer)
-@@ -324,6 +329,7 @@ class DeepcoderDatasetBuilder_Gem(RLDatasetBuilder):
-             format_coef=self.format_coef,
-             reward_timeout=self.reward_timeout,
-             code_qual_every_n=self.code_qual_every_n,
-+            gemini_weight=self.gemini_weight
-         )
-         test_ds = DeepcoderDataset_Gem(
-             batch_size=self.batch_size,
-@@ -335,5 +341,6 @@ class DeepcoderDatasetBuilder_Gem(RLDatasetBuilder):
-             format_coef=self.format_coef,
-             reward_timeout=self.reward_timeout,
-             code_qual_every_n=self.code_qual_every_n,
-+            gemini_weight=self.gemini_weight
-         )
-         return train_ds, test_ds
-diff --git a/tinker_cookbook/recipes/code_rl/gemini_train.py b/tinker_cookbook/recipes/code_rl/gemini_train.py
-index 9283042..df62d7a 100644
---- a/tinker_cookbook/recipes/code_rl/gemini_train.py
-+++ b/tinker_cookbook/recipes/code_rl/gemini_train.py
-@@ -37,6 +37,7 @@ class CLIConfig:
- 
-     # hopefully speeds up training
-     code_qual_every_n: int = 10        # only call Gemini every n episodes
-+    gemini_weight: float=0.2
- 
-     # Logging / eval / checkpoints
-     log_dir: str | None = None
-@@ -71,7 +72,7 @@ async def cli_main(cli_config: CLIConfig) -> None:
- 
-     # Set log path
-     if cli_config.log_path is not None:
--        log_path = cli_config.log_path
-+        log_path = f"{cli_config.log_path}/{run_name}"
-     else:
-         log_path = f"/tmp/tinker-examples/code_rl/{run_name}"
- 
-@@ -84,6 +85,7 @@ async def cli_main(cli_config: CLIConfig) -> None:
-         group_size=cli_config.group_size,
-         seed=cli_config.seed,
-         code_qual_every_n=cli_config.code_qual_every_n,
-+        gemini_weight=cli_config.gemini_weight
-     )
- 
-     config = Config(
-diff --git a/tinker_cookbook/recipes/code_rl/model_metrics/metrics.jsonl b/tinker_cookbook/recipes/code_rl/model_metrics/metrics.jsonl
-index da27d36..ae053db 100644
---- a/tinker_cookbook/recipes/code_rl/model_metrics/metrics.jsonl
-+++ b/tinker_cookbook/recipes/code_rl/model_metrics/metrics.jsonl
-@@ -59,3 +59,62 @@
- {"step": 58, "progress/batch": 58, "optim/lr": 4e-05, "progress/done_frac": 0.15526315789473685, "env/all/ac_tokens_per_turn": 3600.55078125, "env/all/ob_tokens_per_turn": 603.203125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 1843482, "env/all/total_ob_tokens": 308840, "env/all/reward/total": 0.40859375, "env/all/format": 0.984375, "env/all/correct": 0.41015625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.421875, "env/all/by_group/frac_all_good": 0.203125, "env/all/by_group/frac_all_bad": 0.375, "time/assemble_training_data": 3.2540197372436523, "time/train": 105.99414038658142, "optim/kl_sample_train_v1": 0.0007410270627588034, "optim/kl_sample_train_v2": 0.0007277676486410201, "optim/entropy": 0.24245429039001465, "time/compute_kl_sample_train": 0.4762864112854004, "time/save_checkpoint": 3.9526612758636475, "time/total": 499.07599997520447}
- {"step": 59, "progress/batch": 59, "optim/lr": 4e-05, "progress/done_frac": 0.15789473684210525, "env/all/ac_tokens_per_turn": 3736.73828125, "env/all/ob_tokens_per_turn": 668.328125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 1913210, "env/all/total_ob_tokens": 342184, "env/all/reward/total": 0.401953125, "env/all/format": 0.9765625, "env/all/correct": 0.404296875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.375, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.359375, "time/assemble_training_data": 2.2199418544769287, "time/train": 59.657466888427734, "optim/kl_sample_train_v1": 0.0007812347030267119, "optim/kl_sample_train_v2": 0.0009731673053465784, "optim/entropy": 0.24105045199394226, "time/compute_kl_sample_train": 0.36435532569885254, "time/save_checkpoint": 13.924574613571167, "time/total": 830.0928544998169}
- {"step": 60, "progress/batch": 60, "optim/lr": 4e-05, "progress/done_frac": 0.16052631578947368, "test/env/all/ac_tokens_per_turn": 4035.861717612809, "test/env/all/ob_tokens_per_turn": 622.9301310043668, "test/env/all/turns_per_episode": 1.0, "test/env/all/total_episodes": 687, "test/env/all/total_turns": 687, "test/env/all/total_ac_tokens": 2772637, "test/env/all/total_ob_tokens": 427953, "test/env/all/reward/total": 0.29767103347889373, "test/env/all/format": 0.992721979621543, "test/env/all/correct": 0.29839883551673946, "test/env/all/code_quality": 0.0, "test/env/all/by_group/frac_mixed": 0.0, "test/env/all/by_group/frac_all_good": 0.29839883551673946, "test/env/all/by_group/frac_all_bad": 0.7016011644832606, "time/run_evals": 385.3283863067627, "env/all/ac_tokens_per_turn": 4047.439453125, "env/all/ob_tokens_per_turn": 613.890625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2072289, "env/all/total_ob_tokens": 314312, "env/all/reward/total": 0.413671875, "env/all/format": 0.9765625, "env/all/correct": 0.416015625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.390625, "env/all/by_group/frac_all_good": 0.234375, "env/all/by_group/frac_all_bad": 0.375, "time/assemble_training_data": 2.335364580154419, "time/train": 43.2273428440094, "optim/kl_sample_train_v1": 0.0007169598247855902, "optim/kl_sample_train_v2": 0.0007343974430114031, "optim/entropy": 0.23688904941082, "time/compute_kl_sample_train": 0.6435940265655518, "time/save_checkpoint": 4.302887916564941, "time/total": 816.5567348003387}
-+{"step": 61, "progress/batch": 61, "optim/lr": 4e-05, "progress/done_frac": 0.1631578947368421, "env/all/ac_tokens_per_turn": 4100.8671875, "env/all/ob_tokens_per_turn": 602.0625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2099644, "env/all/total_ob_tokens": 308256, "env/all/reward/total": 0.3630859375, "env/all/format": 0.978515625, "env/all/correct": 0.365234375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.375, "env/all/by_group/frac_all_good": 0.1875, "env/all/by_group/frac_all_bad": 0.4375, "time/assemble_training_data": 1.955878734588623, "time/train": 72.78957915306091, "optim/kl_sample_train_v1": 0.0007129737641662359, "optim/kl_sample_train_v2": 0.000757863512262702, "optim/entropy": 0.2500378489494324, "time/compute_kl_sample_train": 0.3935558795928955, "time/save_checkpoint": 5.024366617202759, "time/total": 466.3020634651184}
-+{"step": 62, "progress/batch": 62, "optim/lr": 4e-05, "progress/done_frac": 0.16578947368421051, "env/all/ac_tokens_per_turn": 4124.654296875, "env/all/ob_tokens_per_turn": 633.78125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2111823, "env/all/total_ob_tokens": 324496, "env/all/reward/total": 0.3978515625, "env/all/format": 0.974609375, "env/all/correct": 0.400390625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.53125, "env/all/by_group/frac_all_good": 0.15625, "env/all/by_group/frac_all_bad": 0.3125, "time/assemble_training_data": 2.760097026824951, "time/train": 92.5932719707489, "optim/kl_sample_train_v1": 0.0007485157111659646, "optim/kl_sample_train_v2": 0.0007359634037129581, "optim/entropy": 0.25261491537094116, "time/compute_kl_sample_train": 0.38997364044189453, "time/save_checkpoint": 5.330509901046753, "time/total": 532.3982989788055}
-+{"step": 63, "progress/batch": 63, "optim/lr": 4e-05, "progress/done_frac": 0.16842105263157894, "env/all/ac_tokens_per_turn": 4215.5234375, "env/all/ob_tokens_per_turn": 668.9375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2158348, "env/all/total_ob_tokens": 342496, "env/all/reward/total": 0.3767578125, "env/all/format": 0.978515625, "env/all/correct": 0.37890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.3125, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.421875, "time/assemble_training_data": 2.073240280151367, "time/train": 39.37544822692871, "optim/kl_sample_train_v1": 0.000765854783821851, "optim/kl_sample_train_v2": 0.0007583317346870899, "optim/entropy": 0.25587353110313416, "time/compute_kl_sample_train": 0.407517671585083, "time/save_checkpoint": 4.39504075050354, "time/total": 478.7189950942993}
-+{"step": 64, "progress/batch": 64, "optim/lr": 4e-05, "progress/done_frac": 0.17105263157894737, "env/all/ac_tokens_per_turn": 3800.30859375, "env/all/ob_tokens_per_turn": 656.671875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 1945758, "env/all/total_ob_tokens": 336216, "env/all/reward/total": 0.43359375, "env/all/format": 0.98046875, "env/all/correct": 0.435546875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.1875, "env/all/by_group/frac_all_bad": 0.34375, "time/assemble_training_data": 1.8805177211761475, "time/train": 58.895509243011475, "optim/kl_sample_train_v1": 0.0007291344227269292, "optim/kl_sample_train_v2": 0.0007277238764800131, "optim/entropy": 0.2366204559803009, "time/compute_kl_sample_train": 0.39226508140563965, "time/save_checkpoint": 5.231722831726074, "time/total": 465.22417402267456}
-+{"step": 65, "progress/batch": 65, "optim/lr": 4e-05, "progress/done_frac": 0.1736842105263158, "env/all/ac_tokens_per_turn": 3928.91015625, "env/all/ob_tokens_per_turn": 607.6875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2011602, "env/all/total_ob_tokens": 311136, "env/all/reward/total": 0.5533203125, "env/all/format": 0.986328125, "env/all/correct": 0.5546875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.375, "env/all/by_group/frac_all_good": 0.359375, "env/all/by_group/frac_all_bad": 0.265625, "time/assemble_training_data": 3.51012921333313, "time/train": 94.48019409179688, "optim/kl_sample_train_v1": 0.0008431168971583247, "optim/kl_sample_train_v2": 0.000757030094973743, "optim/entropy": 0.2518639862537384, "time/compute_kl_sample_train": 0.3787569999694824, "time/save_checkpoint": 10.061316728591919, "time/total": 503.47350454330444}
-+{"step": 66, "progress/batch": 66, "optim/lr": 4e-05, "progress/done_frac": 0.1763157894736842, "env/all/ac_tokens_per_turn": 4111.68359375, "env/all/ob_tokens_per_turn": 599.296875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2105182, "env/all/total_ob_tokens": 306840, "env/all/reward/total": 0.502734375, "env/all/format": 0.96875, "env/all/correct": 0.505859375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.421875, "env/all/by_group/frac_all_good": 0.328125, "env/all/by_group/frac_all_bad": 0.25, "time/assemble_training_data": 2.0607900619506836, "time/train": 54.005473375320435, "optim/kl_sample_train_v1": 0.0007392195402644575, "optim/kl_sample_train_v2": 0.0007269275956787169, "optim/entropy": 0.2378670871257782, "time/compute_kl_sample_train": 0.39116358757019043, "time/save_checkpoint": 4.81585955619812, "time/total": 468.90008664131165}
-+{"step": 67, "progress/batch": 67, "optim/lr": 4e-05, "progress/done_frac": 0.17894736842105263, "env/all/ac_tokens_per_turn": 4463.232421875, "env/all/ob_tokens_per_turn": 639.96875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2285175, "env/all/total_ob_tokens": 327664, "env/all/reward/total": 0.4228515625, "env/all/format": 0.990234375, "env/all/correct": 0.423828125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.34375, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.390625, "time/assemble_training_data": 2.6910579204559326, "time/train": 72.78628516197205, "optim/kl_sample_train_v1": 0.0007519930368289351, "optim/kl_sample_train_v2": 0.0007719504646956921, "optim/entropy": 0.2614273726940155, "time/compute_kl_sample_train": 0.47204136848449707, "time/save_checkpoint": 4.2209553718566895, "time/total": 663.9875729084015}
-+{"step": 68, "progress/batch": 68, "optim/lr": 4e-05, "progress/done_frac": 0.18157894736842106, "env/all/ac_tokens_per_turn": 5193.216796875, "env/all/ob_tokens_per_turn": 623.625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2658927, "env/all/total_ob_tokens": 319296, "env/all/reward/total": 0.3404296875, "env/all/format": 0.966796875, "env/all/correct": 0.34375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.390625, "env/all/by_group/frac_all_good": 0.1875, "env/all/by_group/frac_all_bad": 0.421875, "time/assemble_training_data": 2.7149651050567627, "time/train": 108.91330289840698, "optim/kl_sample_train_v1": 0.0007425922085531056, "optim/kl_sample_train_v2": 0.0007843142375349998, "optim/entropy": 0.256545752286911, "time/compute_kl_sample_train": 0.5138335227966309, "time/save_checkpoint": 3.6596665382385254, "time/total": 605.2298347949982}
-+{"step": 69, "progress/batch": 69, "optim/lr": 4e-05, "progress/done_frac": 0.18421052631578946, "env/all/ac_tokens_per_turn": 4106.7109375, "env/all/ob_tokens_per_turn": 640.65625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2102636, "env/all/total_ob_tokens": 328016, "env/all/reward/total": 0.46660156249999996, "env/all/format": 0.978515625, "env/all/correct": 0.46875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.4375, "env/all/by_group/frac_all_good": 0.28125, "env/all/by_group/frac_all_bad": 0.28125, "time/assemble_training_data": 2.354677438735962, "time/train": 46.46767520904541, "optim/kl_sample_train_v1": 0.0007292740046977997, "optim/kl_sample_train_v2": 0.0007592625333927572, "optim/entropy": 0.25384584069252014, "time/compute_kl_sample_train": 0.43543148040771484, "time/save_checkpoint": 5.1600306034088135, "time/total": 500.31817603111267}
-+{"step": 70, "progress/batch": 70, "optim/lr": 4e-05, "progress/done_frac": 0.1868421052631579, "env/all/ac_tokens_per_turn": 4772.416015625, "env/all/ob_tokens_per_turn": 621.78125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2443477, "env/all/total_ob_tokens": 318352, "env/all/reward/total": 0.46035156250000003, "env/all/format": 0.955078125, "env/all/correct": 0.46484375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.453125, "env/all/by_group/frac_all_good": 0.234375, "env/all/by_group/frac_all_bad": 0.3125, "time/assemble_training_data": 4.036370038986206, "time/train": 91.42486310005188, "optim/kl_sample_train_v1": 0.0007744911126792431, "optim/kl_sample_train_v2": 0.000785319076385349, "optim/entropy": 0.25661277770996094, "time/compute_kl_sample_train": 0.5051913261413574, "time/save_checkpoint": 9.939294815063477, "time/total": 527.1561625003815}
-+{"step": 71, "progress/batch": 71, "optim/lr": 4e-05, "progress/done_frac": 0.18947368421052632, "env/all/ac_tokens_per_turn": 5139.658203125, "env/all/ob_tokens_per_turn": 667.59375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2631505, "env/all/total_ob_tokens": 341808, "env/all/reward/total": 0.36562500000000003, "env/all/format": 0.96484375, "env/all/correct": 0.369140625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.375, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.40625, "time/assemble_training_data": 2.8248019218444824, "time/train": 52.80256390571594, "optim/kl_sample_train_v1": 0.0007642004638910294, "optim/kl_sample_train_v2": 0.000798879424110055, "optim/entropy": 0.2507244348526001, "time/compute_kl_sample_train": 0.5112433433532715, "time/save_checkpoint": 6.424597263336182, "time/total": 864.3110725879669}
-+{"step": 72, "progress/batch": 72, "optim/lr": 4e-05, "progress/done_frac": 0.19210526315789472, "env/all/ac_tokens_per_turn": 4921.544921875, "env/all/ob_tokens_per_turn": 631.453125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2519831, "env/all/total_ob_tokens": 323304, "env/all/reward/total": 0.4103515625, "env/all/format": 0.962890625, "env/all/correct": 0.4140625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.421875, "env/all/by_group/frac_all_good": 0.234375, "env/all/by_group/frac_all_bad": 0.34375, "time/assemble_training_data": 2.770724058151245, "time/train": 79.27884817123413, "optim/kl_sample_train_v1": 0.0007340014562942088, "optim/kl_sample_train_v2": 0.0007543033570982516, "optim/entropy": 0.24883754551410675, "time/compute_kl_sample_train": 0.45674967765808105, "time/save_checkpoint": 10.269767999649048, "time/total": 513.0691194534302}
-+{"step": 73, "progress/batch": 73, "optim/lr": 4e-05, "progress/done_frac": 0.19473684210526315, "env/all/ac_tokens_per_turn": 4491.9453125, "env/all/ob_tokens_per_turn": 570.140625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2299876, "env/all/total_ob_tokens": 291912, "env/all/reward/total": 0.459375, "env/all/format": 0.96484375, "env/all/correct": 0.462890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.359375, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.34375, "time/assemble_training_data": 3.33052396774292, "time/train": 45.20476031303406, "optim/kl_sample_train_v1": 0.000714560505002737, "optim/kl_sample_train_v2": 0.0007241423591040075, "optim/entropy": 0.23401479423046112, "time/compute_kl_sample_train": 0.4406709671020508, "time/save_checkpoint": 5.352816581726074, "time/total": 425.33766984939575}
-+{"step": 74, "progress/batch": 74, "optim/lr": 4e-05, "progress/done_frac": 0.19736842105263158, "env/all/ac_tokens_per_turn": 4843.40234375, "env/all/ob_tokens_per_turn": 690.359375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2479822, "env/all/total_ob_tokens": 353464, "env/all/reward/total": 0.42421875, "env/all/format": 0.96484375, "env/all/correct": 0.427734375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.34375, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.359375, "time/assemble_training_data": 3.311959981918335, "time/train": 79.14457297325134, "optim/kl_sample_train_v1": 0.000807260163128376, "optim/kl_sample_train_v2": 0.0008908911258913577, "optim/entropy": 0.25892263650894165, "time/compute_kl_sample_train": 0.48247456550598145, "time/save_checkpoint": 5.734542369842529, "time/total": 466.3489010334015}
-+{"step": 75, "progress/batch": 75, "optim/lr": 4e-05, "progress/done_frac": 0.2, "env/all/ac_tokens_per_turn": 5327.181640625, "env/all/ob_tokens_per_turn": 677.234375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2727517, "env/all/total_ob_tokens": 346744, "env/all/reward/total": 0.5666015625, "env/all/format": 0.962890625, "env/all/correct": 0.5703125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.53125, "env/all/by_group/frac_all_good": 0.328125, "env/all/by_group/frac_all_bad": 0.140625, "time/assemble_training_data": 2.5860674381256104, "time/train": 121.16189885139465, "optim/kl_sample_train_v1": 0.0007557867211289704, "optim/kl_sample_train_v2": 0.0007840506732463837, "optim/entropy": 0.26095670461654663, "time/compute_kl_sample_train": 0.5610935688018799, "time/save_checkpoint": 3.9387595653533936, "time/total": 596.6014330387115}
-+{"step": 76, "progress/batch": 76, "optim/lr": 4e-05, "progress/done_frac": 0.2026315789473684, "env/all/ac_tokens_per_turn": 5064.21484375, "env/all/ob_tokens_per_turn": 633.015625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2592878, "env/all/total_ob_tokens": 324104, "env/all/reward/total": 0.4537109375, "env/all/format": 0.947265625, "env/all/correct": 0.458984375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.265625, "time/assemble_training_data": 3.9661757946014404, "time/train": 69.49004697799683, "optim/kl_sample_train_v1": 0.0007578496588394046, "optim/kl_sample_train_v2": 0.0007650575134903193, "optim/entropy": 0.2414751797914505, "time/compute_kl_sample_train": 0.40097951889038086, "time/save_checkpoint": 4.722691059112549, "time/total": 715.7278115749359}
-+{"step": 77, "progress/batch": 77, "optim/lr": 4e-05, "progress/done_frac": 0.20526315789473684, "env/all/ac_tokens_per_turn": 5195.056640625, "env/all/ob_tokens_per_turn": 615.796875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2659869, "env/all/total_ob_tokens": 315288, "env/all/reward/total": 0.472265625, "env/all/format": 0.95703125, "env/all/correct": 0.4765625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.28125, "time/assemble_training_data": 2.809549331665039, "time/train": 107.65428733825684, "optim/kl_sample_train_v1": 0.0007406935328617692, "optim/kl_sample_train_v2": 0.0007393347332254052, "optim/entropy": 0.2369009256362915, "time/compute_kl_sample_train": 0.5096969604492188, "time/save_checkpoint": 4.055797100067139, "time/total": 635.4810333251953}
-+{"step": 78, "progress/batch": 78, "optim/lr": 4e-05, "progress/done_frac": 0.20789473684210527, "env/all/ac_tokens_per_turn": 5966.369140625, "env/all/ob_tokens_per_turn": 633.609375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3054781, "env/all/total_ob_tokens": 324408, "env/all/reward/total": 0.4751953125, "env/all/format": 0.947265625, "env/all/correct": 0.48046875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.265625, "time/assemble_training_data": 3.6181554794311523, "time/train": 57.35490965843201, "optim/kl_sample_train_v1": 0.0008083676802925766, "optim/kl_sample_train_v2": 0.0008221598691307008, "optim/entropy": 0.26470455527305603, "time/compute_kl_sample_train": 0.6646840572357178, "time/save_checkpoint": 5.744938135147095, "time/total": 435.008492231369}
-+{"step": 79, "progress/batch": 79, "optim/lr": 4e-05, "progress/done_frac": 0.21052631578947367, "env/all/ac_tokens_per_turn": 5114.806640625, "env/all/ob_tokens_per_turn": 662.1875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2618781, "env/all/total_ob_tokens": 339040, "env/all/reward/total": 0.50546875, "env/all/format": 0.95703125, "env/all/correct": 0.509765625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.484375, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.21875, "time/assemble_training_data": 2.651275396347046, "time/train": 92.87135934829712, "optim/kl_sample_train_v1": 0.0007618505624122918, "optim/kl_sample_train_v2": 0.0007744884351268411, "optim/entropy": 0.2540193796157837, "time/compute_kl_sample_train": 0.5104160308837891, "time/save_checkpoint": 11.396274328231812, "time/total": 480.4909133911133}
-+{"step": 80, "progress/batch": 80, "optim/lr": 4e-05, "progress/done_frac": 0.2131578947368421, "test/env/all/ac_tokens_per_turn": 6020.71615720524, "test/env/all/ob_tokens_per_turn": 622.9301310043668, "test/env/all/turns_per_episode": 1.0, "test/env/all/total_episodes": 687, "test/env/all/total_turns": 687, "test/env/all/total_ac_tokens": 4136232, "test/env/all/total_ob_tokens": 427953, "test/env/all/reward/total": 0.34687045123726346, "test/env/all/format": 0.9898107714701602, "test/env/all/correct": 0.34788937409024745, "test/env/all/code_quality": 0.0, "test/env/all/by_group/frac_mixed": 0.0, "test/env/all/by_group/frac_all_good": 0.34788937409024745, "test/env/all/by_group/frac_all_bad": 0.6521106259097526, "time/run_evals": 822.1428515911102, "env/all/ac_tokens_per_turn": 5273.46875, "env/all/ob_tokens_per_turn": 582.859375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2700016, "env/all/total_ob_tokens": 298424, "env/all/reward/total": 0.40605468749999996, "env/all/format": 0.958984375, "env/all/correct": 0.41015625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.421875, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.359375, "time/assemble_training_data": 4.126157522201538, "time/train": 54.09513330459595, "optim/kl_sample_train_v1": 0.0007920349598862231, "optim/kl_sample_train_v2": 0.0007961852243170142, "optim/entropy": 0.2649328410625458, "time/compute_kl_sample_train": 0.47695326805114746, "time/save_checkpoint": 7.115128040313721, "time/total": 1324.176747083664}
-+{"step": 81, "progress/batch": 81, "optim/lr": 4e-05, "progress/done_frac": 0.21578947368421053, "env/all/ac_tokens_per_turn": 5748.380859375, "env/all/ob_tokens_per_turn": 610.390625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2943171, "env/all/total_ob_tokens": 312520, "env/all/reward/total": 0.458203125, "env/all/format": 0.953125, "env/all/correct": 0.462890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.59375, "env/all/by_group/frac_all_good": 0.1875, "env/all/by_group/frac_all_bad": 0.21875, "time/assemble_training_data": 2.8800933361053467, "time/train": 88.82266306877136, "optim/kl_sample_train_v1": 0.0007255392847582698, "optim/kl_sample_train_v2": 0.0007545968983322382, "optim/entropy": 0.2502441108226776, "time/compute_kl_sample_train": 0.5525960922241211, "time/save_checkpoint": 12.087316036224365, "time/total": 1006.5410113334656}
-+{"step": 82, "progress/batch": 82, "optim/lr": 4e-05, "progress/done_frac": 0.21842105263157896, "env/all/ac_tokens_per_turn": 5699.248046875, "env/all/ob_tokens_per_turn": 635.0625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2918015, "env/all/total_ob_tokens": 325152, "env/all/reward/total": 0.36386718749999997, "env/all/format": 0.947265625, "env/all/correct": 0.369140625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.484375, "env/all/by_group/frac_all_good": 0.171875, "env/all/by_group/frac_all_bad": 0.34375, "time/assemble_training_data": 2.9548239707946777, "time/train": 59.52150082588196, "optim/kl_sample_train_v1": 0.0007119643851183355, "optim/kl_sample_train_v2": 0.0007409076206386089, "optim/entropy": 0.2393849641084671, "time/compute_kl_sample_train": 0.5457124710083008, "time/save_checkpoint": 5.283956527709961, "time/total": 750.710280418396}
-+{"step": 83, "progress/batch": 83, "optim/lr": 4e-05, "progress/done_frac": 0.22105263157894736, "env/all/ac_tokens_per_turn": 5183.537109375, "env/all/ob_tokens_per_turn": 596.8125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2653971, "env/all/total_ob_tokens": 305568, "env/all/reward/total": 0.46328125, "env/all/format": 0.96484375, "env/all/correct": 0.466796875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.390625, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.3125, "time/assemble_training_data": 2.5172691345214844, "time/train": 84.13076519966125, "optim/kl_sample_train_v1": 0.0008152179070748389, "optim/kl_sample_train_v2": 0.0011330855777487159, "optim/entropy": 0.2701372802257538, "time/compute_kl_sample_train": 0.48560357093811035, "time/save_checkpoint": 12.072690725326538, "time/total": 661.9946753978729}
-+{"step": 84, "progress/batch": 84, "optim/lr": 4e-05, "progress/done_frac": 0.2236842105263158, "env/all/ac_tokens_per_turn": 6301.595703125, "env/all/ob_tokens_per_turn": 610.296875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3226417, "env/all/total_ob_tokens": 312472, "env/all/reward/total": 0.3591796875, "env/all/format": 0.939453125, "env/all/correct": 0.365234375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.40625, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.375, "time/assemble_training_data": 3.317223310470581, "time/train": 49.83653736114502, "optim/kl_sample_train_v1": 0.0007953907479532063, "optim/kl_sample_train_v2": 0.000809243880212307, "optim/entropy": 0.2562867999076843, "time/compute_kl_sample_train": 0.6260726451873779, "time/save_checkpoint": 6.388319730758667, "time/total": 611.0197877883911}
-+{"step": 85, "progress/batch": 85, "optim/lr": 4e-05, "progress/done_frac": 0.22631578947368422, "env/all/ac_tokens_per_turn": 5816.251953125, "env/all/ob_tokens_per_turn": 592.78125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2977921, "env/all/total_ob_tokens": 303504, "env/all/reward/total": 0.44101562499999997, "env/all/format": 0.95703125, "env/all/correct": 0.4453125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.234375, "time/assemble_training_data": 3.050915479660034, "time/train": 93.02502703666687, "optim/kl_sample_train_v1": 0.0007938620983622968, "optim/kl_sample_train_v2": 0.0008242496405728161, "optim/entropy": 0.25299039483070374, "time/compute_kl_sample_train": 0.5620284080505371, "time/save_checkpoint": 6.276669502258301, "time/total": 818.876168012619}
-+{"step": 86, "progress/batch": 86, "optim/lr": 4e-05, "progress/done_frac": 0.22894736842105262, "env/all/ac_tokens_per_turn": 6893.111328125, "env/all/ob_tokens_per_turn": 657.671875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3529273, "env/all/total_ob_tokens": 336728, "env/all/reward/total": 0.4392578125, "env/all/format": 0.919921875, "env/all/correct": 0.447265625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.59375, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.1875, "time/assemble_training_data": 3.529449939727783, "time/train": 66.44688081741333, "optim/kl_sample_train_v1": 0.0008126916945911944, "optim/kl_sample_train_v2": 0.0007843044004403055, "optim/entropy": 0.2481648325920105, "time/compute_kl_sample_train": 0.5681817531585693, "time/save_checkpoint": 4.757361650466919, "time/total": 650.7923405170441}
-+{"step": 87, "progress/batch": 87, "optim/lr": 4e-05, "progress/done_frac": 0.23157894736842105, "env/all/ac_tokens_per_turn": 4740.458984375, "env/all/ob_tokens_per_turn": 631.828125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2427115, "env/all/total_ob_tokens": 323496, "env/all/reward/total": 0.4337890625, "env/all/format": 0.982421875, "env/all/correct": 0.435546875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.328125, "env/all/by_group/frac_all_good": 0.28125, "env/all/by_group/frac_all_bad": 0.390625, "time/assemble_training_data": 2.3946151733398438, "time/train": 96.9060869216919, "optim/kl_sample_train_v1": 0.0007655090885236859, "optim/kl_sample_train_v2": 0.0007674454827792943, "optim/entropy": 0.24569962918758392, "time/compute_kl_sample_train": 0.3980569839477539, "time/save_checkpoint": 6.916744709014893, "time/total": 585.7020690441132}
-+{"step": 88, "progress/batch": 88, "optim/lr": 4e-05, "progress/done_frac": 0.23421052631578948, "env/all/ac_tokens_per_turn": 5126.892578125, "env/all/ob_tokens_per_turn": 599.921875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2624969, "env/all/total_ob_tokens": 307160, "env/all/reward/total": 0.480078125, "env/all/format": 0.9765625, "env/all/correct": 0.482421875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5, "env/all/by_group/frac_all_good": 0.25, "env/all/by_group/frac_all_bad": 0.25, "time/assemble_training_data": 3.267496347427368, "time/train": 61.5831937789917, "optim/kl_sample_train_v1": 0.0007726757903583348, "optim/kl_sample_train_v2": 0.0007877484895288944, "optim/entropy": 0.2445507049560547, "time/compute_kl_sample_train": 0.4686729907989502, "time/save_checkpoint": 14.010720491409302, "time/total": 531.815149307251}
-+{"step": 89, "progress/batch": 89, "optim/lr": 4e-05, "progress/done_frac": 0.23684210526315788, "env/all/ac_tokens_per_turn": 6131.30078125, "env/all/ob_tokens_per_turn": 587.75, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3139226, "env/all/total_ob_tokens": 300928, "env/all/reward/total": 0.4775390625, "env/all/format": 0.970703125, "env/all/correct": 0.48046875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.453125, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.28125, "time/assemble_training_data": 3.274540424346924, "time/train": 115.05573439598083, "optim/kl_sample_train_v1": 0.0008398364880122244, "optim/kl_sample_train_v2": 0.0008278980385512114, "optim/entropy": 0.2597050070762634, "time/compute_kl_sample_train": 0.5843901634216309, "time/save_checkpoint": 4.841803789138794, "time/total": 587.3787033557892}
-+{"step": 90, "progress/batch": 90, "optim/lr": 4e-05, "progress/done_frac": 0.2394736842105263, "env/all/ac_tokens_per_turn": 7321.7109375, "env/all/ob_tokens_per_turn": 607.5625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3748716, "env/all/total_ob_tokens": 311072, "env/all/reward/total": 0.451171875, "env/all/format": 0.94140625, "env/all/correct": 0.45703125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.265625, "time/assemble_training_data": 8.99599003791809, "time/train": 71.91230487823486, "optim/kl_sample_train_v1": 0.0008222148171626031, "optim/kl_sample_train_v2": 0.000802700815256685, "optim/entropy": 0.25993815064430237, "time/compute_kl_sample_train": 0.9058628082275391, "time/save_checkpoint": 4.593205213546753, "time/total": 562.0443263053894}
-+{"step": 91, "progress/batch": 91, "optim/lr": 4e-05, "progress/done_frac": 0.24210526315789474, "env/all/ac_tokens_per_turn": 7725.125, "env/all/ob_tokens_per_turn": 683.9375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3955264, "env/all/total_ob_tokens": 350176, "env/all/reward/total": 0.2984375, "env/all/format": 0.91796875, "env/all/correct": 0.306640625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.4375, "env/all/by_group/frac_all_good": 0.15625, "env/all/by_group/frac_all_bad": 0.40625, "time/assemble_training_data": 3.821619749069214, "time/train": 126.69476342201233, "optim/kl_sample_train_v1": 0.0008644174668006599, "optim/kl_sample_train_v2": 0.000847010815050453, "optim/entropy": 0.2600173354148865, "time/compute_kl_sample_train": 0.6107897758483887, "time/save_checkpoint": 4.470027923583984, "time/total": 604.3270461559296}
-+{"step": 92, "progress/batch": 92, "optim/lr": 4e-05, "progress/done_frac": 0.24473684210526317, "env/all/ac_tokens_per_turn": 5912.75, "env/all/ob_tokens_per_turn": 617.90625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3027328, "env/all/total_ob_tokens": 316368, "env/all/reward/total": 0.4591796875, "env/all/format": 0.962890625, "env/all/correct": 0.462890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.40625, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.328125, "time/assemble_training_data": 3.111407518386841, "time/train": 77.21666812896729, "optim/kl_sample_train_v1": 0.0007759092259220779, "optim/kl_sample_train_v2": 0.0008018516818992794, "optim/entropy": 0.25052717328071594, "time/compute_kl_sample_train": 0.4959895610809326, "time/save_checkpoint": 4.125257253646851, "time/total": 493.90431785583496}
-+{"step": 93, "progress/batch": 93, "optim/lr": 4e-05, "progress/done_frac": 0.24736842105263157, "env/all/ac_tokens_per_turn": 6221.740234375, "env/all/ob_tokens_per_turn": 636.984375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3185531, "env/all/total_ob_tokens": 326136, "env/all/reward/total": 0.5001953125, "env/all/format": 0.962890625, "env/all/correct": 0.50390625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.328125, "env/all/by_group/frac_all_good": 0.375, "env/all/by_group/frac_all_bad": 0.296875, "time/assemble_training_data": 3.2602477073669434, "time/train": 121.37819290161133, "optim/kl_sample_train_v1": 0.0007730340003035963, "optim/kl_sample_train_v2": 0.000783971103373915, "optim/entropy": 0.25721853971481323, "time/compute_kl_sample_train": 0.5967426300048828, "time/save_checkpoint": 3.9007010459899902, "time/total": 540.8940420150757}
-+{"step": 94, "progress/batch": 94, "optim/lr": 4e-05, "progress/done_frac": 0.25, "env/all/ac_tokens_per_turn": 6161.470703125, "env/all/ob_tokens_per_turn": 598.59375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3154673, "env/all/total_ob_tokens": 306480, "env/all/reward/total": 0.44375, "env/all/format": 0.9453125, "env/all/correct": 0.44921875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.40625, "env/all/by_group/frac_all_good": 0.3125, "env/all/by_group/frac_all_bad": 0.28125, "time/assemble_training_data": 3.135960578918457, "time/train": 75.81186747550964, "optim/kl_sample_train_v1": 0.0008215093403123319, "optim/kl_sample_train_v2": 0.000911459734197706, "optim/entropy": 0.262195885181427, "time/compute_kl_sample_train": 0.4760715961456299, "time/save_checkpoint": 4.697887420654297, "time/total": 495.1736443042755}
-+{"step": 95, "progress/batch": 95, "optim/lr": 4e-05, "progress/done_frac": 0.25263157894736843, "env/all/ac_tokens_per_turn": 5981.470703125, "env/all/ob_tokens_per_turn": 610.0, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3062513, "env/all/total_ob_tokens": 312320, "env/all/reward/total": 0.48750000000000004, "env/all/format": 0.97265625, "env/all/correct": 0.490234375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.390625, "env/all/by_group/frac_all_good": 0.28125, "env/all/by_group/frac_all_bad": 0.328125, "time/assemble_training_data": 3.6912291049957275, "time/train": 121.80239224433899, "optim/kl_sample_train_v1": 0.0008140887948684394, "optim/kl_sample_train_v2": 0.0008393992320634425, "optim/entropy": 0.25484567880630493, "time/compute_kl_sample_train": 0.5712270736694336, "time/save_checkpoint": 5.239192485809326, "time/total": 526.8879718780518}
-+{"step": 96, "progress/batch": 96, "optim/lr": 4e-05, "progress/done_frac": 0.25526315789473686, "env/all/ac_tokens_per_turn": 7159.34375, "env/all/ob_tokens_per_turn": 649.5625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3665584, "env/all/total_ob_tokens": 332576, "env/all/reward/total": 0.375390625, "env/all/format": 0.9453125, "env/all/correct": 0.380859375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.515625, "env/all/by_group/frac_all_good": 0.203125, "env/all/by_group/frac_all_bad": 0.28125, "time/assemble_training_data": 3.936847686767578, "time/train": 103.44382953643799, "optim/kl_sample_train_v1": 0.0008215652778744698, "optim/kl_sample_train_v2": 0.0008382444502785802, "optim/entropy": 0.2562706172466278, "time/compute_kl_sample_train": 0.6600625514984131, "time/save_checkpoint": 5.643333435058594, "time/total": 790.4159741401672}
-+{"step": 97, "progress/batch": 97, "optim/lr": 4e-05, "progress/done_frac": 0.2578947368421053, "env/all/ac_tokens_per_turn": 5544.978515625, "env/all/ob_tokens_per_turn": 624.53125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2839029, "env/all/total_ob_tokens": 319760, "env/all/reward/total": 0.4775390625, "env/all/format": 0.970703125, "env/all/correct": 0.48046875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5625, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.21875, "time/assemble_training_data": 3.6626930236816406, "time/train": 119.38676261901855, "optim/kl_sample_train_v1": 0.0007535387412644923, "optim/kl_sample_train_v2": 0.0008040162501856685, "optim/entropy": 0.24431905150413513, "time/compute_kl_sample_train": 0.4201827049255371, "time/save_checkpoint": 4.734087228775024, "time/total": 1659.5557315349579}
-+{"step": 98, "progress/batch": 98, "optim/lr": 4e-05, "progress/done_frac": 0.26052631578947366, "env/all/ac_tokens_per_turn": 6738.79296875, "env/all/ob_tokens_per_turn": 593.25, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3450262, "env/all/total_ob_tokens": 303744, "env/all/reward/total": 0.394140625, "env/all/format": 0.95703125, "env/all/correct": 0.3984375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.453125, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.328125, "time/assemble_training_data": 3.3571531772613525, "time/train": 104.67119240760803, "optim/kl_sample_train_v1": 0.0008664170163683593, "optim/kl_sample_train_v2": 0.0008574793464504182, "optim/entropy": 0.2590669095516205, "time/compute_kl_sample_train": 0.5645139217376709, "time/save_checkpoint": 11.195820093154907, "time/total": 2185.0190246105194}
-+{"step": 99, "progress/batch": 99, "optim/lr": 4e-05, "progress/done_frac": 0.2631578947368421, "env/all/ac_tokens_per_turn": 7189.3828125, "env/all/ob_tokens_per_turn": 747.21875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3680964, "env/all/total_ob_tokens": 382576, "env/all/reward/total": 0.493359375, "env/all/format": 0.93359375, "env/all/correct": 0.5, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.578125, "env/all/by_group/frac_all_good": 0.234375, "env/all/by_group/frac_all_bad": 0.1875, "time/assemble_training_data": 5.5760650634765625, "time/train": 110.28515243530273, "optim/kl_sample_train_v1": 0.0008380688959732652, "optim/kl_sample_train_v2": 0.000847597315441817, "optim/entropy": 0.26120951771736145, "time/compute_kl_sample_train": 0.8077473640441895, "time/save_checkpoint": 37.43190383911133, "time/total": 895.887617111206}
-+{"step": 100, "progress/batch": 100, "optim/lr": 4e-05, "progress/done_frac": 0.2657894736842105, "test/env/all/ac_tokens_per_turn": 8704.46288209607, "test/env/all/ob_tokens_per_turn": 622.9301310043668, "test/env/all/turns_per_episode": 1.0, "test/env/all/total_episodes": 687, "test/env/all/total_turns": 687, "test/env/all/total_ac_tokens": 5979966, "test/env/all/total_ob_tokens": 427953, "test/env/all/reward/total": 0.3739446870451237, "test/env/all/format": 0.9548762736535662, "test/env/all/correct": 0.3784570596797671, "test/env/all/code_quality": 0.0, "test/env/all/by_group/frac_mixed": 0.0, "test/env/all/by_group/frac_all_good": 0.3784570596797671, "test/env/all/by_group/frac_all_bad": 0.6215429403202329, "time/run_evals": 881.861668586731, "env/all/ac_tokens_per_turn": 7349.306640625, "env/all/ob_tokens_per_turn": 693.09375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3762845, "env/all/total_ob_tokens": 354864, "env/all/reward/total": 0.5570312500000001, "env/all/format": 0.9453125, "env/all/correct": 0.5625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5625, "env/all/by_group/frac_all_good": 0.3125, "env/all/by_group/frac_all_bad": 0.125, "time/assemble_training_data": 4.02209997177124, "time/train": 98.76641631126404, "optim/kl_sample_train_v1": 0.000850596756208688, "optim/kl_sample_train_v2": 0.0008978616097010672, "optim/entropy": 0.2662164866924286, "time/compute_kl_sample_train": 0.7538068294525146, "time/save_checkpoint": 4.266815900802612, "time/total": 1807.6481301784515}
-+{"step": 101, "progress/batch": 101, "optim/lr": 4e-05, "progress/done_frac": 0.26842105263157895, "env/all/ac_tokens_per_turn": 5651.705078125, "env/all/ob_tokens_per_turn": 636.6875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 2893673, "env/all/total_ob_tokens": 325984, "env/all/reward/total": 0.5662109375, "env/all/format": 0.978515625, "env/all/correct": 0.568359375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.421875, "env/all/by_group/frac_all_good": 0.328125, "env/all/by_group/frac_all_bad": 0.25, "time/assemble_training_data": 3.365229606628418, "time/train": 97.91362929344177, "optim/kl_sample_train_v1": 0.0007940888754092157, "optim/kl_sample_train_v2": 0.0008139540441334248, "optim/entropy": 0.25089189410209656, "time/compute_kl_sample_train": 0.5446500778198242, "time/save_checkpoint": 3.981137275695801, "time/total": 1590.399689912796}
-+{"step": 102, "progress/batch": 102, "optim/lr": 4e-05, "progress/done_frac": 0.2710526315789474, "env/all/ac_tokens_per_turn": 7795.484375, "env/all/ob_tokens_per_turn": 601.390625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3991288, "env/all/total_ob_tokens": 307912, "env/all/reward/total": 0.448828125, "env/all/format": 0.9375, "env/all/correct": 0.455078125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.453125, "env/all/by_group/frac_all_good": 0.28125, "env/all/by_group/frac_all_bad": 0.265625, "time/assemble_training_data": 4.004104852676392, "time/train": 83.63651633262634, "optim/kl_sample_train_v1": 0.0008891591569408774, "optim/kl_sample_train_v2": 0.0009713858016766608, "optim/entropy": 0.2748545706272125, "time/compute_kl_sample_train": 0.7395839691162109, "time/save_checkpoint": 9.295893907546997, "time/total": 2589.246472597122}
-+{"step": 103, "progress/batch": 103, "optim/lr": 4e-05, "progress/done_frac": 0.2736842105263158, "env/all/ac_tokens_per_turn": 7042.9609375, "env/all/ob_tokens_per_turn": 579.546875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3605996, "env/all/total_ob_tokens": 296728, "env/all/reward/total": 0.38203125000000004, "env/all/format": 0.953125, "env/all/correct": 0.38671875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.375, "env/all/by_group/frac_all_good": 0.25, "env/all/by_group/frac_all_bad": 0.375, "time/assemble_training_data": 3.513632297515869, "time/train": 102.2714421749115, "optim/kl_sample_train_v1": 0.0008245595963671803, "optim/kl_sample_train_v2": 0.000859438325278461, "optim/entropy": 0.2606046497821808, "time/compute_kl_sample_train": 0.6424887180328369, "time/save_checkpoint": 3.9541585445404053, "time/total": 686.9463911056519}
-+{"step": 104, "progress/batch": 104, "optim/lr": 4e-05, "progress/done_frac": 0.27631578947368424, "env/all/ac_tokens_per_turn": 7288.294921875, "env/all/ob_tokens_per_turn": 666.546875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3731607, "env/all/total_ob_tokens": 341272, "env/all/reward/total": 0.4947265625, "env/all/format": 0.927734375, "env/all/correct": 0.501953125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.578125, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.15625, "time/assemble_training_data": 3.956955671310425, "time/train": 117.03522777557373, "optim/kl_sample_train_v1": 0.0008275998407043517, "optim/kl_sample_train_v2": 0.0009102640324272215, "optim/entropy": 0.2545662522315979, "time/compute_kl_sample_train": 0.6717512607574463, "time/save_checkpoint": 3.9251043796539307, "time/total": 685.8004202842712}
-+{"step": 105, "progress/batch": 105, "optim/lr": 4e-05, "progress/done_frac": 0.2789473684210526, "env/all/ac_tokens_per_turn": 6509.32421875, "env/all/ob_tokens_per_turn": 597.53125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3332774, "env/all/total_ob_tokens": 305936, "env/all/reward/total": 0.5240234375, "env/all/format": 0.966796875, "env/all/correct": 0.52734375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.453125, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.25, "time/assemble_training_data": 4.1241981983184814, "time/train": 76.99196529388428, "optim/kl_sample_train_v1": 0.0008079506224021316, "optim/kl_sample_train_v2": 0.0008453584741801023, "optim/entropy": 0.25429171323776245, "time/compute_kl_sample_train": 0.6094574928283691, "time/save_checkpoint": 6.553384065628052, "time/total": 635.4741232395172}
-+{"step": 106, "progress/batch": 106, "optim/lr": 4e-05, "progress/done_frac": 0.28157894736842104, "env/all/ac_tokens_per_turn": 7438.494140625, "env/all/ob_tokens_per_turn": 635.390625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3808509, "env/all/total_ob_tokens": 325320, "env/all/reward/total": 0.4560546875, "env/all/format": 0.931640625, "env/all/correct": 0.462890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.484375, "env/all/by_group/frac_all_good": 0.21875, "env/all/by_group/frac_all_bad": 0.296875, "time/assemble_training_data": 3.644308567047119, "time/train": 133.0656225681305, "optim/kl_sample_train_v1": 0.0008811148582026362, "optim/kl_sample_train_v2": 0.0008604304166510701, "optim/entropy": 0.26179957389831543, "time/compute_kl_sample_train": 0.7015194892883301, "time/save_checkpoint": 4.221517324447632, "time/total": 690.050662279129}
-+{"step": 107, "progress/batch": 107, "optim/lr": 4e-05, "progress/done_frac": 0.28421052631578947, "env/all/ac_tokens_per_turn": 7429.787109375, "env/all/ob_tokens_per_turn": 603.640625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3804051, "env/all/total_ob_tokens": 309064, "env/all/reward/total": 0.5740234375, "env/all/format": 0.958984375, "env/all/correct": 0.578125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.34375, "env/all/by_group/frac_all_bad": 0.1875, "time/assemble_training_data": 4.51533055305481, "time/train": 111.48648643493652, "optim/kl_sample_train_v1": 0.0008376098121516407, "optim/kl_sample_train_v2": 0.0008334123995155096, "optim/entropy": 0.2553633749485016, "time/compute_kl_sample_train": 0.6786785125732422, "time/save_checkpoint": 3.9903922080993652, "time/total": 884.5821073055267}
-+{"step": 108, "progress/batch": 108, "optim/lr": 4e-05, "progress/done_frac": 0.2868421052631579, "env/all/ac_tokens_per_turn": 7247.09765625, "env/all/ob_tokens_per_turn": 664.8125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3710514, "env/all/total_ob_tokens": 340384, "env/all/reward/total": 0.48066406249999993, "env/all/format": 0.962890625, "env/all/correct": 0.484375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5, "env/all/by_group/frac_all_good": 0.234375, "env/all/by_group/frac_all_bad": 0.265625, "time/assemble_training_data": 4.259993076324463, "time/train": 63.262934923172, "optim/kl_sample_train_v1": 0.0008760808850638568, "optim/kl_sample_train_v2": 0.0008724471554160118, "optim/entropy": 0.26400184631347656, "time/compute_kl_sample_train": 0.5234494209289551, "time/save_checkpoint": 4.114532470703125, "time/total": 556.9325280189514}
-+{"step": 109, "progress/batch": 109, "optim/lr": 4e-05, "progress/done_frac": 0.2894736842105263, "env/all/ac_tokens_per_turn": 6601.455078125, "env/all/ob_tokens_per_turn": 628.484375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3379945, "env/all/total_ob_tokens": 321784, "env/all/reward/total": 0.626953125, "env/all/format": 0.9609375, "env/all/correct": 0.630859375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.4375, "env/all/by_group/frac_all_good": 0.453125, "env/all/by_group/frac_all_bad": 0.109375, "time/assemble_training_data": 3.622040033340454, "time/train": 117.40883612632751, "optim/kl_sample_train_v1": 0.0008536537061445415, "optim/kl_sample_train_v2": 0.0010574560146778822, "optim/entropy": 0.25133076310157776, "time/compute_kl_sample_train": 0.5227105617523193, "time/save_checkpoint": 3.874939441680908, "time/total": 547.3552525043488}
-+{"step": 110, "progress/batch": 110, "optim/lr": 4e-05, "progress/done_frac": 0.29210526315789476, "env/all/ac_tokens_per_turn": 8477.640625, "env/all/ob_tokens_per_turn": 615.75, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 4340552, "env/all/total_ob_tokens": 315264, "env/all/reward/total": 0.4931640625, "env/all/format": 0.951171875, "env/all/correct": 0.498046875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.234375, "time/assemble_training_data": 4.45789361000061, "time/train": 90.43608021736145, "optim/kl_sample_train_v1": 0.0008522261632606387, "optim/kl_sample_train_v2": 0.0008485540747642517, "optim/entropy": 0.2594420611858368, "time/compute_kl_sample_train": 0.8349332809448242, "time/save_checkpoint": 10.797351121902466, "time/total": 551.5252871513367}
-+{"step": 111, "progress/batch": 111, "optim/lr": 4e-05, "progress/done_frac": 0.29473684210526313, "env/all/ac_tokens_per_turn": 7388.41796875, "env/all/ob_tokens_per_turn": 618.109375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3782870, "env/all/total_ob_tokens": 316472, "env/all/reward/total": 0.40546875, "env/all/format": 0.953125, "env/all/correct": 0.41015625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.4375, "env/all/by_group/frac_all_good": 0.25, "env/all/by_group/frac_all_bad": 0.3125, "time/assemble_training_data": 6.321980714797974, "time/train": 67.80576848983765, "optim/kl_sample_train_v1": 0.0008204769110307097, "optim/kl_sample_train_v2": 0.0008323087822645903, "optim/entropy": 0.2652226984500885, "time/compute_kl_sample_train": 0.8219287395477295, "time/save_checkpoint": 3.9112792015075684, "time/total": 530.6068513393402}
-+{"step": 112, "progress/batch": 112, "optim/lr": 4e-05, "progress/done_frac": 0.29736842105263156, "env/all/ac_tokens_per_turn": 7397.900390625, "env/all/ob_tokens_per_turn": 574.0, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3787725, "env/all/total_ob_tokens": 293888, "env/all/reward/total": 0.5810546875, "env/all/format": 0.931640625, "env/all/correct": 0.587890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.375, "env/all/by_group/frac_all_bad": 0.15625, "time/assemble_training_data": 3.7538435459136963, "time/train": 144.7019865512848, "optim/kl_sample_train_v1": 0.0008324794471263885, "optim/kl_sample_train_v2": 0.0008618704741820693, "optim/entropy": 0.2568267583847046, "time/compute_kl_sample_train": 0.7635910511016846, "time/save_checkpoint": 12.283966779708862, "time/total": 569.8595714569092}
-+{"step": 113, "progress/batch": 113, "optim/lr": 4e-05, "progress/done_frac": 0.3, "env/all/ac_tokens_per_turn": 8515.54296875, "env/all/ob_tokens_per_turn": 616.1875, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 4359958, "env/all/total_ob_tokens": 315488, "env/all/reward/total": 0.4470703125, "env/all/format": 0.939453125, "env/all/correct": 0.453125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5, "env/all/by_group/frac_all_good": 0.25, "env/all/by_group/frac_all_bad": 0.25, "time/assemble_training_data": 4.816187381744385, "time/train": 74.45352721214294, "optim/kl_sample_train_v1": 0.0008693161071278155, "optim/kl_sample_train_v2": 0.0009513375116512179, "optim/entropy": 0.27028536796569824, "time/compute_kl_sample_train": 0.8865134716033936, "time/save_checkpoint": 4.792590379714966, "time/total": 546.4363391399384}
-+{"step": 114, "progress/batch": 114, "optim/lr": 4e-05, "progress/done_frac": 0.3026315789473684, "env/all/ac_tokens_per_turn": 6837.900390625, "env/all/ob_tokens_per_turn": 627.578125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3501005, "env/all/total_ob_tokens": 321320, "env/all/reward/total": 0.576953125, "env/all/format": 0.94921875, "env/all/correct": 0.58203125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.453125, "env/all/by_group/frac_all_good": 0.40625, "env/all/by_group/frac_all_bad": 0.140625, "time/assemble_training_data": 3.4635772705078125, "time/train": 59.99387454986572, "optim/kl_sample_train_v1": 0.0007773558609187603, "optim/kl_sample_train_v2": 0.0007963377283886075, "optim/entropy": 0.2410389482975006, "time/compute_kl_sample_train": 0.6311531066894531, "time/save_checkpoint": 3.831674098968506, "time/total": 564.771892786026}
-+{"step": 115, "progress/batch": 115, "optim/lr": 4e-05, "progress/done_frac": 0.30526315789473685, "env/all/ac_tokens_per_turn": 7749.0390625, "env/all/ob_tokens_per_turn": 575.15625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3967508, "env/all/total_ob_tokens": 294480, "env/all/reward/total": 0.5296875000000001, "env/all/format": 0.92578125, "env/all/correct": 0.537109375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.484375, "env/all/by_group/frac_all_good": 0.3125, "env/all/by_group/frac_all_bad": 0.203125, "time/assemble_training_data": 4.6967363357543945, "time/train": 117.45049405097961, "optim/kl_sample_train_v1": 0.0008358832565136254, "optim/kl_sample_train_v2": 0.0008907660376280546, "optim/entropy": 0.26068052649497986, "time/compute_kl_sample_train": 0.7042226791381836, "time/save_checkpoint": 4.8226847648620605, "time/total": 568.671971321106}
-+{"step": 116, "progress/batch": 116, "optim/lr": 4e-05, "progress/done_frac": 0.3078947368421053, "env/all/ac_tokens_per_turn": 8546.82421875, "env/all/ob_tokens_per_turn": 637.84375, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 4375974, "env/all/total_ob_tokens": 326576, "env/all/reward/total": 0.46191406250000006, "env/all/format": 0.931640625, "env/all/correct": 0.46875, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.46875, "env/all/by_group/frac_all_good": 0.296875, "env/all/by_group/frac_all_bad": 0.234375, "time/assemble_training_data": 4.182077884674072, "time/train": 80.10848045349121, "optim/kl_sample_train_v1": 0.0008333830046467483, "optim/kl_sample_train_v2": 0.0008614315302111208, "optim/entropy": 0.2561946213245392, "time/compute_kl_sample_train": 0.7842161655426025, "time/save_checkpoint": 4.407613515853882, "time/total": 534.7334151268005}
-+{"step": 117, "progress/batch": 117, "optim/lr": 4e-05, "progress/done_frac": 0.3105263157894737, "env/all/ac_tokens_per_turn": 7488.62109375, "env/all/ob_tokens_per_turn": 609.65625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3834174, "env/all/total_ob_tokens": 312144, "env/all/reward/total": 0.4576171875, "env/all/format": 0.947265625, "env/all/correct": 0.462890625, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.4375, "env/all/by_group/frac_all_good": 0.265625, "env/all/by_group/frac_all_bad": 0.296875, "time/assemble_training_data": 4.364379644393921, "time/train": 133.66431283950806, "optim/kl_sample_train_v1": 0.0008436625357717276, "optim/kl_sample_train_v2": 0.0008187596686184406, "optim/entropy": 0.2536691725254059, "time/compute_kl_sample_train": 0.661395788192749, "time/save_checkpoint": 3.9337804317474365, "time/total": 554.7914309501648}
-+{"step": 118, "progress/batch": 118, "optim/lr": 4e-05, "progress/done_frac": 0.3131578947368421, "env/all/ac_tokens_per_turn": 8197.517578125, "env/all/ob_tokens_per_turn": 599.140625, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 4197129, "env/all/total_ob_tokens": 306760, "env/all/reward/total": 0.3265625, "env/all/format": 0.9453125, "env/all/correct": 0.33203125, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.5, "env/all/by_group/frac_all_good": 0.140625, "env/all/by_group/frac_all_bad": 0.359375, "time/assemble_training_data": 4.9159650802612305, "time/train": 102.49077939987183, "optim/kl_sample_train_v1": 0.0008162789163179696, "optim/kl_sample_train_v2": 0.0009127316880039871, "optim/entropy": 0.2639845013618469, "time/compute_kl_sample_train": 0.7195086479187012, "time/save_checkpoint": 3.6368191242218018, "time/total": 566.2343838214874}
-+{"step": 119, "progress/batch": 119, "optim/lr": 4e-05, "progress/done_frac": 0.3157894736842105, "env/all/ac_tokens_per_turn": 7179.759765625, "env/all/ob_tokens_per_turn": 602.953125, "env/all/turns_per_episode": 1.0, "env/all/total_episodes": 512, "env/all/total_turns": 512, "env/all/total_ac_tokens": 3676037, "env/all/total_ob_tokens": 308712, "env/all/reward/total": 0.48574218750000003, "env/all/format": 0.955078125, "env/all/correct": 0.490234375, "env/all/code_quality": 0.0, "env/all/by_group/frac_mixed": 0.359375, "env/all/by_group/frac_all_good": 0.328125, "env/all/by_group/frac_all_bad": 0.3125, "time/assemble_training_data": 3.763838529586792, "time/train": 115.75839829444885, "optim/kl_sample_train_v1": 0.0008523390279151499, "optim/kl_sample_train_v2": 0.0008612835663370788, "optim/entropy": 0.2568064033985138, "time/compute_kl_sample_train": 0.6638331413269043, "time/save_checkpoint": 11.519805431365967, "time/total": 511.22725200653076}
diff --git a/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/config.json b/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/config.json
deleted file mode 100644
index 93eaeaf..0000000
--- a/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/config.json
+++ /dev/null
@@ -1,37 +0,0 @@
-{
-  "learning_rate": 4e-05,
-  "dataset_builder": {
-    "batch_size": 64,
-    "model_name_for_tokenizer": "Qwen/Qwen3-4B-Instruct-2507",
-    "renderer_name": "qwen3_instruct",
-    "group_size": 8,
-    "convo_prefix": null,
-    "seed": 0,
-    "format_coef": 0.1,
-    "reward_timeout": 6,
-    "code_qual_every_n": 10,
-    "gemini_weight": 0.2
-  },
-  "model_name": "Qwen/Qwen3-4B-Instruct-2507",
-  "max_tokens": 24576,
-  "temperature": 1.0,
-  "compute_post_kl": false,
-  "evaluator_builders": [],
-  "lora_rank": 32,
-  "kl_penalty_coef": 0.0,
-  "kl_discount_factor": 0.0,
-  "loss_fn": "importance_sampling",
-  "num_substeps": 1,
-  "wandb_project": "gemini_code_gen",
-  "wandb_name": "deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02",
-  "log_path": "gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02",
-  "base_url": null,
-  "enable_trace": false,
-  "remove_constant_reward_groups": false,
-  "eval_every": 20,
-  "save_every": 20,
-  "load_checkpoint_path": null,
-  "async_config": null,
-  "stream_minibatch_config": null,
-  "num_groups_to_log": 4
-}
\ No newline at end of file
diff --git a/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/logs.log b/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/logs.log
deleted file mode 100644
index ef4ed34..0000000
--- a/gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02/logs.log
+++ /dev/null
@@ -1,3 +0,0 @@
-tinker_cookbook.utils.ml_log:490 [INFO] Logging to: gemini_code_gen_two_tenths_weight\deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02
-tinker_cookbook.checkpoint_utils:21 [INFO] No checkpoints found at gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02\checkpoints.jsonl
-tinker_cookbook.checkpoint_utils:52 [INFO] No checkpoints found with key state_path in gemini_code_gen_two_tenths_weight/deepcoder-Qwen-Qwen3-4B-Instruct-2507-32rank-4e-05lr-8group-64batch-seed0-2025-12-06-22-02
